### PR TITLE
feat: Added auth and basic routing

### PR DESCRIPTION
**Pull request by Hardiegogo**
Added auth and basic routing:
- created auth context
- used mockbee for auth
- added routing to notes page after login

Preview-link: https://deploy-preview-1--zippy-seahorse-51d18a.netlify.app/

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { useAuth } from "./context/useAuth";
+import NotesListing from "./pages/notes-page/NotesListing";
+
+function App() {
+  const {authState}=useAuth()
+  const {isUserActive}=authState
+  return (
+    <div className="App">
+      <Routes>
+        <Route path='/mock' element={<Mockman/>}/>
```
Comment (parthgabani777) [inline src/App.js:16]: you can add route in another files

File: src/components/navbar-component/Navbar.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            <div className="header-right">
+                <nav>
+                    <ul>
+                        <li>
+                            {authState.isUserActive?<button onClick={()=>logoutHandler(dispatchAuth,navigate)} ><i class='bx bx-log-out-circle icons'></i></button>:<Link to='/login'><i className='bx bx-log-in-circle icons'></i></Link>}
+                        </li>
+                        <li>
+                            <div className="badge-wrapper">
+                                {/* <button onClick={()=>wishlistClickHandler(authState,navigate)}><i className='bx bx-heart icons'></i></button>
+                                <div className="badge btn-badge">{cartState.activeUserWishList.length}</div> */}
```
Comment (parthgabani777) [inline src/components/navbar-component/Navbar.jsx:39]: you can remove comment

**Review by parthgabani777 — APPROVED** [review]
code looks good!

File: src/context/useAuth.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        case "USER_LOGIN":
+            return {
+                ...state,
+                isUserActive:true,
+                userDetails:action.payload
+            }
+        case "USER_LOGOUT":
+            localStorage.removeItem('token');
+            return{
+                ...state,
```
Comment (rahulyadav139) [inline src/context/useAuth.js:29]: Codes look clean and readable great work.

**Review by rahulyadav139 — APPROVED** [review]

File: src/App.js (hunk 1 above)
Comment (Hardiegogo) [reply src/App.js:16]: Okay

File: src/components/navbar-component/Navbar.jsx (hunk 2 above)
Comment (Hardiegogo) [reply src/components/navbar-component/Navbar.jsx:39]: yes I will do so.

**Pull request merged**